src/rules/interfaceNameRule.ts

```diff
@@ -45,8 +53,34 @@ class NameWalker extends Lint.RuleWalker {
         return (firstCharacter === "I");
     }
 
-    private addFailureAt(position: number, width: number) {
-        const failure = this.createFailure(position, width, Rule.FAILURE_STRING);
+    private hasPrefixI(name: string): boolean {
+        if (name.length <= 0) {
+            return true;
+        }
+
+        const firstCharacter = name.charAt(0);
+        if (firstCharacter !== "I") {
+            return false;
+        }
+
+        const secondCharacter = name.charAt(1);
+        if (!secondCharacter) {
```

Comment:
We prefer to use explicit comparisons for non-boolean conditions, for example here:
`secondCharacter === ""`